### PR TITLE
feat: [CCM-8098]: Added new types for CCM UI

### PR DESCRIPTION
**Pull request by jenil-harness**
### Summary
Added custom prop types for CCM UI child app.

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by jenil-harness** [comment]
retrigger titlecheck

**Review by rishik30 — DISMISSED** [review]

**Review by rishik30 — DISMISSED** [review]

**Review by vkbansal — DISMISSED** [review]

File: src/microfrontends/README.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# Microfrontend Types
+
+This package is used in Next Gen UI and across microfrontends for type definations.
```
Comment (abhinavrastogi-harness) [inline src/microfrontends/README.md:3]: spelling of `definitions`

File: src/microfrontends/README.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+### How to publish types?
+
+1. Make the required changes.
+2. Bump up the version in `src/microfrontends/package.json` according to changes made. Please follow [SemVer](https://semver.org/).
+3. Get it merged to develop branch.
+4. Package will be autopublished once the changes are merged to develop branch.
+   > Note: For development purposes publish packages with `alpha`, `beta` or corresponding tags.
```
[the remark this answers is not in the document]
Comment (abhinavrastogi-harness) [reply src/microfrontends/README.md:11]: and publish a beta version only if necessary

**Comment by jenil-harness** [comment]
retrigger coverage

**Review by abhinavrastogi-harness — APPROVED** [review]

**Review by rishik30 — APPROVED** [review]

**Pull request merged**